lib/Conversion/CalyxToHW/CalyxToHW.cpp

```diff
@@ -310,6 +310,20 @@ struct ConvertCellOp : public OpInterfaceConversionPattern<CellInterface> {
           auto wire = wireIn(op.in(), op.instanceName(), "", b);
           wires.append({wire.input(), wire});
         })
+        .Case([&](PadLibOp op) {
+          auto in = wireIn(op.in(), op.instanceName(), op.portName(op.in()), b);
+          auto srcWidth = in.getType().getIntOrFloatBitWidth();
+          auto destWidth = op.out().getType().getIntOrFloatBitWidth();
+          auto zero = b.create<hw::ConstantOp>(op.getLoc(),
+                                               APInt(destWidth - srcWidth, 0));
+          auto padded = b.createOrFold<comb::ConcatOp>(zero, in);
```

Comment:
Why not call `wireOut` on `padded`? All the other comb conversions have their result passed through a wire that may or may not be canonicalized away, and I think it might be necessary here as well.